src/datasource.js

```diff
@@ -50,17 +52,12 @@ export class GenericDatasource {
 
     return this.q.all(promises).then(richResponses => {
       var result = _.map(richResponses, (richResponse) => {
-        var response = richResponse.response;
-        var datapoints;
-        if (response.data.length != 0) {
-          datapoints = _.map(response.data[0].data, point => [point.value, point.timestamp]);
-        } else {
-          datapoints = [];
-        }
         return {
           refId: richResponse.refId,
           target: richResponse.target,
-          datapoints: datapoints
+          // The javascript's flatMap
+          datapoints: [].concat.apply([], richResponse.response.data.map(d => d.data))
```

Comment:
And btw, I changed the result process in case of error it now looks at the result status rather than the content of "data". If it's not 200, then empty array is returned.